src/odfesqlodbc/es_communication.cpp

```diff
@@ -438,9 +444,98 @@ int ESCommunication::ExecDirect(const char* query, const char* fetch_size_) {
         return -1;
     }
     m_result_queue.push(std::unique_ptr< ESResult >(result));
+    if (result->cursor.size() > 0) {
+        auto send_cursor_queries =
+            std::async(std::launch::async, [&]() {
```

Comment:
So I believe what' you're doing here is launching a thread to retrieve results, then another thread to process the results? Can you please add some comments to the code with descriptions as to what is going on?